examples/src/main/java/com/mapbox/navigation/examples/core/RouteRefreshActivity.kt

```diff
@@ -0,0 +1,484 @@
+                    .center(point)
+                    .zoom(13.0)
+                    .build()
+                mapboxMap.setCamera(cameraOptions)
+                mapboxNavigation.unregisterLocationObserver(this)
+            }
+
+            override fun onNewLocationMatcherResult(
+                locationMatcherResult: LocationMatcherResult,
+            ) {
+                // not handled
+            }
+        })
+
+        // initialize Navigation Camera
+        viewportDataSource = MapboxNavigationViewportDataSource(
+            binding.mapView.getMapboxMap()
+        )
+        navigationCamera = NavigationCamera(
+            binding.mapView.getMapboxMap(),
+            binding.mapView.camera,
+            viewportDataSource
+        )
+        binding.mapView.camera.addCameraAnimationsLifecycleListener(
+            NavigationBasicGesturesHandler(navigationCamera)
+        )
+        navigationCamera.registerNavigationCameraStateChangeObserver { navigationCameraState ->
+            // shows/hide the recenter button depending on the camera state
+            when (navigationCameraState) {
+                NavigationCameraState.TRANSITION_TO_FOLLOWING,
+                NavigationCameraState.FOLLOWING -> binding.recenter.visibility = INVISIBLE
+
+                NavigationCameraState.TRANSITION_TO_OVERVIEW,
+                NavigationCameraState.OVERVIEW,
+                NavigationCameraState.IDLE -> binding.recenter.visibility = VISIBLE
+            }
+        }
+        if (this.resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE) {
+            viewportDataSource.overviewPadding = landscapeOverviewPadding
+        } else {
+            viewportDataSource.overviewPadding = overviewPadding
+        }
+        if (this.resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE) {
+            viewportDataSource.followingPadding = landscapeFollowingPadding
+        } else {
+            viewportDataSource.followingPadding = followingPadding
+        }
+
+        // initialize route line
+        val mapboxRouteLineOptions = MapboxRouteLineOptions.Builder(this)
+            .withRouteLineBelowLayerId("road-label")
+            .withRouteLineResources(
+                RouteLineResources.Builder()
+                    .routeLineColorResources(
+                        RouteLineColorResources.Builder()
+                            .alternativeRouteDefaultColor(Color.parseColor("#FF0000"))
+                            .alternativeRouteHeavyCongestionColor(Color.parseColor("#00FF00"))
+                            .alternativeRouteLowCongestionColor(Color.parseColor("#0000FF"))
+                            .alternativeRouteModerateCongestionColor(Color.parseColor("#FFFF00"))
+                            .alternativeRouteSevereCongestionColor(Color.parseColor("#FF00FF"))
+                            .alternativeRouteUnknownCongestionColor(Color.parseColor("#00FFFF"))
```

Comment:
Could you use washed/greyed out green, yellow, red, crimson, black instead? Same can be done for the primary route without washing/greying the colors out. That's going to be much easier to understand at a glance.

Reply:
I'll cut a separate PR for this.

Reply:
Do you want specific colours for specific values? If so, what's corresponding to what?
But actually does it really matter? I mean with this example you immediately see what was refreshed (because after you clear it, the whole line just becomes single-coloured). Does it matter what traffic it represents? In my understanding that would not be the use case for this example but maybe you have one.

Reply:
```
unknown -> blue
low -> green
moderate -> yellow
heavy -> red
severe -> crimson
closed -> black
```
this would allow the example to be more flexible (for example to also test correctness) and easier to understand at a glance for anyone. The colors right now don't correspond to anything you might be familiar with so the verification is binary - refreshes or doesn't - while in practice it's more nuanced to see whether correct segments were refreshed and whether they have correct values and the above color schema would make that immediately clear without having to go into the code to see which color means what.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/6858